packages/ui/src/views/repo/repo-commits/components/filter.tsx

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from 'react'
+
+import { ListActions, SearchBox } from '@/components'
+import { isEmpty } from 'lodash-es'
+
+import { useCommonFilter } from '../hooks/useCommonFilter'
+
+interface FilterProps {
+  showSort?: boolean
+  sortOptions?: ListActions.DropdownItemProps[]
+  showSearch?: boolean
+}
+
+const Filter = <S,>({ showSort, sortOptions, showSearch = true }: FilterProps) => {
+  const { sort, query, handleSearch, handleDropdownChange } = useCommonFilter<S>()
```

Comment:
we don't want to use `useCommonFilter` anymore. Let's remove it.
also, the sort/query etc should be maintained in gitness app via `nuqs`, and we just need to pass the props and callbacks to the view.